harness/determined/cli/experiment.py

```diff
@@ -304,66 +320,99 @@ def describe(args: Namespace) -> None:
-                    + [
-                        checkpoint_state,
-                        render.format_time(checkpoint_end_time),
-                        validation_state,
-                        render.format_time(validation_end_time),
-                    ]
-                    + v_metrics_fields
-                )
-                values.append(row)
+                    v_metrics_fields = [None for name in v_metrics_names]
+
+                if step_detail:
+                    if step_detail.totalBatches in step_output:
+                        # condense training, checkpoints, validation workloads into one 'step' row
+                        # for compatibility with previous versions of describe
+                        merge_row = step_output[step_detail.totalBatches]
+                        merge_row[3] = max(merge_row[3], render.format_time(step_detail.endTime))
+                        for idx, tfield in enumerate(t_metrics_fields):
+                            if tfield and merge_row[4 + idx] is None:
+                                merge_row[4 + idx] = tfield
+                        start_checkpoint = 4 + len(t_metrics_fields)
+                        if checkpoint_state:
+                            merge_row[start_checkpoint] = checkpoint_state.replace("STATE_", "")
+                            merge_row[start_checkpoint + 1] = render.format_time(
+                                checkpoint_end_time
+                            )
+                        if validation_end_time:
+                            merge_row[start_checkpoint + 3] = render.format_time(
+                                validation_end_time
+                            )
+                        if validation_state:
+                            merge_row[start_checkpoint + 2] = validation_state.replace("STATE_", "")
+                        for idx, vfield in enumerate(v_metrics_fields):
+                            if vfield and merge_row[start_checkpoint + idx + 4] is None:
+                                merge_row[start_checkpoint + idx + 4] = vfield
+                    else:
+                        row = (
+                            [
+                                trial.id,
+                                step_detail.totalBatches,
+                                step_detail.state.value.replace("STATE_", ""),
+                                render.format_time(step_detail.endTime),
+                            ]
+                            + t_metrics_fields
+                            + [
+                                checkpoint_state,
+                                render.format_time(checkpoint_end_time),
+                                validation_state,
+                                render.format_time(validation_end_time),
+                            ]
+                            + v_metrics_fields
+                        )
+                        step_output[step_detail.totalBatches] = row
```

Comment:
if our tests allow it we might want to go away from displaying the information in a step package all together. It'd be a breaking change but it's a direction we want to go in and could reduce the changes needed here. (If you wanna go that route I'd check with the team first on this)